### DISCUSSION TITLE
add vue and apache echarts

### DISCUSSION THREAD

**Discussion by cmdares** [TotallyInformation/node-red-contrib-uibuilder#115] [General]
It is suggested to use Vue as the basic framework and echarts as the overall charts control. 
Ecarts has rich components and many control parameters. 
I hope the community can upgrade.
Thanks.

**Comment by TotallyInformation** [comment]
Hi, thanks for engaging.

But why is this an issue?

The default installation of uibuilder already includes VueJS but uibuilder absolutely doesn't (and never will) _require_ Vue.

uibuilder is designed to work with any (or indeed no) framework quite happily. The inclusion of Vue is merely a convenience for people with limited experience of front-end coding as it lets you get started very quickly and easily.

As for a charting package - most people will not need one so it certainly will never be included in the default installation. You can easily install whatever you want.

I've moved this to a discussion since it isn't an issue with uibuilder.
